diff --git a/lib/parse.js b/lib/parse.js
index 3a6988e629f52aee6609dc5ec290a5e9ff47cecb..533f78b7fe01641233d786814b1246229745ce1a 100644
--- a/lib/parse.js
+++ b/lib/parse.js
@@ -144,6 +144,9 @@ const parse = (input, options = {}) => {
 
     if (value === CHAR_LEFT_PARENTHESES) {
       block = push({ type: 'paren', nodes: [] });
+      if (stack.length > 128) {
+        throw new SyntaxError('Input nesting exceeds maximum depth of 128');
+      }
       stack.push(block);
       push({ type: 'text', value });
       continue;
@@ -210,6 +213,9 @@ const parse = (input, options = {}) => {
       };
 
       block = push(brace);
+      if (stack.length > 128) {
+        throw new SyntaxError('Input nesting exceeds maximum depth of 128');
+      }
       stack.push(block);
       push({ type: 'open', value });
       continue;
