diff --git a/index.js b/index.js
index 56c26025074cbcf218e3ec4d810662da93f8024e..e63ec855b6e73bcdb98615b1f15cc2cb5b5012de 100644
--- a/index.js
+++ b/index.js
@@ -396,7 +396,7 @@ module.exports = class CachePolicy {
         this._assertRequestHasHeaders(req);
 
         // In all circumstances, a cache MUST NOT ignore the must-revalidate directive
-        if (this._rescc['must-revalidate']) {
+        if (this.maxAge() === 0 || this._rescc['must-revalidate']) {
             return this._evaluateRequestMissResult(req);
         }
 
